lib/model/folder.go

```diff
@@ -1023,8 +1036,19 @@ func (f *folder) monitorWatch(ctx context.Context) {
 			}
 			aggrCancel()
 			errChan = nil
+			if summarySub != nil {
+				summarySub.Unsubscribe()
+			}
 			aggrCtx, aggrCancel = context.WithCancel(ctx)
+		case summary := <-summaryChan:
+			if !f.warnedKqueue && summary.Data.(map[string]interface{})["localTotalItems"].(int) > kqueueItemCountThreshold {
```

Comment:
Seems like we should have defined types for the event data probably

Reply:
Not should we have done that in general, I also should check the type. This would have panicked, as we wrap the map into another map. I added the types and a failure condition if the type is wrong.